test/ansible/run_test.yaml

```diff
@@ -10,15 +10,26 @@
     example_manifest: "{{ lookup('ansible.builtin.env', 'EXAMPLE_MANIFEST', default='multi-node-example.yaml') }}"
     number_of_nodes: "{{ lookup('ansible.builtin.env', 'NUMBER_OF_NODES', default='7') }}"
 
-    capi_version: v1.9.4
-    capm3_version: v1.9.2
+    capi_version: "{{ lookup('ansible.builtin.env', 'CAPI_VERSION', default='v1.9.4') }}" 
+    capm3_version: "{{ lookup('ansible.builtin.env', 'CAPM3_VERSION', default='v1.9.2') }}"  
     dist_dir: "{{ lookup('ansible.builtin.env', 'DIST_DIR') }}"
     src_dir: /tmp/capbcoa
     kind_cluster_name: capi-baremetal-provider
     ssh_authorized_key: "{{ lookup('ansible.builtin.env', 'SSH_AUTHORIZED_KEY') }}"
     pullsecret: "{{ lookup('ansible.builtin.env', 'PULLSECRET') }}"
     container_tag: "{{ lookup('ansible.builtin.env', 'CONTAINER_TAG', default='local')}}"
   tasks:
+  - name: prepare infrastructure-operator
+    delegate_to: localhost
+    shell: |
+      envsubst < "{{ playbook_dir }}/../e2e/manifests/infrastructure-operator/kustomization.yaml" > kustomization.yaml.new
```

Comment:
nit: we migth want to save this as `/e2e/manifests/infrastructure-operator/kustomization.yaml.tpl`, so it's clear that it's not an usable kustomization file, and we would need only one step here.

Probably we could do the same with other files, but it'd be outside the scope of this PR.

Reply:
Also, we might want to do this in the target machine instead, so we can leave local copy clean